src/routes/controllers/tag.controller.js

```diff
@@ -0,0 +1,104 @@
+import sequelize from '../../../.sequelizerc';
+import models from '../../db/models';
+import {
+  successResponse,
+  errorResponse,
+  verifyToken,
+} from '../../utils/helpers.utils';
+
+const { ArticleTags, Tags } = models;
+
+/**
+ * @description create tag for an article
+ * @param {*} tags
+ * @param {*} articleId
+ * @returns {boolean} response
+ */
+export async function createTag(tags, articleId) {
+  if (!Array.isArray(tags)) return null;
+  let createdTag = [];
+  for (let tag of tags) {
```

Comment:
We can use list map function to iterate through the list
```javascript
tags.map(tag => {
    const response = await processTags(tag.toLowerCase(), articleId);
    if (response.status === 'error') return response.message;
    createdTag.push(response.message);
});
```

Reply:
I have tried the recommendation using `.map` and `.forEach`. but I have one common issue with both of them, I can't retrieve the `response message`. 
When I used the `forEach`, I had to create a variable that stores the error message on every loop.
```
The issue I have with this is that I have to wait until it has processed all the tags in the loop and stores the message in the error message, then checks if the variable is not **null/undefined** and throw an error.
```